File: android/core/src/main/kotlin/com/coldopen/core/QuoteCorpus.kt
package com.coldopen.core

import org.json.JSONArray
import java.io.InputStream

/**
 * Parses the quote corpus. This module has no Android dependency on purpose
 * (see build.gradle.kts) — it takes raw JSON in and hands `Quote`s back, so
 * it stays testable with plain `gradle :core:test` and a JDK, no Android SDK
 * or emulator required. The `:app` module is responsible for supplying the
 * actual bundled corpus (from `app/src/main/assets/quotes.json`, synced via
 * `npm run sync:android` at the repo root).
 */
object QuoteCorpus {
    fun parse(json: String): List<Quote> {
        val array = JSONArray(json)
        return (0 until array.length()).map { i ->
            val obj = array.getJSONObject(i)
            Quote(
                text = obj.getString("text"),
                episodeTitle = obj.getString("episodeTitle"),
                season = obj.getInt("season"),
                episode = obj.getInt("episode"),
                arc = obj.getString("arc"),
            )
        }
    }

    fun load(stream: InputStream): List<Quote> =
        parse(stream.bufferedReader().use { it.readText() })
}

File: windows/WidgetProvider.Core/QuoteCorpus.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace ColdOpen.WidgetProvider.Core;

/// <summary>
/// Parses the quote corpus. Takes raw JSON in, hands <see cref="Quote"/>s
/// back — no dependency on where the JSON came from, so the widget
/// provider project is responsible for actually locating and reading
/// <c>quotes.json</c> (see WidgetProvider/WidgetProvider.cs).
/// </summary>
public static class QuoteCorpus
{
    private sealed record QuoteDto(
        [property: JsonPropertyName("text")] string Text,
        [property: JsonPropertyName("episodeTitle")] string EpisodeTitle,
        [property: JsonPropertyName("season")] int Season,
        [property: JsonPropertyName("episode")] int Episode,
        [property: JsonPropertyName("arc")] string Arc);

    public static IReadOnlyList<Quote> Parse(string json)
    {
        var dtos = JsonSerializer.Deserialize<List<QuoteDto>>(json)
            ?? throw new InvalidDataException("Corpus JSON did not deserialize to a list.");

        return dtos
            .Select(d => new Quote(d.Text, d.EpisodeTitle, d.Season, d.Episode, d.Arc))
            .ToList();
    }

    public static IReadOnlyList<Quote> Load(string path) => Parse(File.ReadAllText(path));
}
